src/modules/mavlink/mavlink_ftp.cpp

```diff
@@ -361,11 +361,12 @@ MavlinkFTP::_workList(PayloadHeader *payload, bool list_hidden)
 #endif
 			// For files we get the file size as well
 			direntType = kDirentFile;
-			snprintf(buf, sizeof(buf), "%s/%s", dirPath, result->d_name);
-			struct stat st;
+			if (snprintf(buf, sizeof(buf), "%s/%s", dirPath, result->d_name)) {
```

Comment:
Shouldn't we compare against `sizeof(buf)`? See http://www.cplusplus.com/reference/cstdio/snprintf/